File: packages/native-ios/wrst-runtime/Sources/WrstRuntime/AppConfig.swift
import SwiftUI
import Combine

// Mirrors AppConfig.kt - app-wide config set from JS via createAppConfig().
// Background color lives in @Published state so the root view updates on reload.
@MainActor
final class AppConfig: ObservableObject {
    static let shared = AppConfig()

    private static let defaultBackground = Color.black

    @Published var backgroundColor: Color = AppConfig.defaultBackground
    // Whether the NavigationStack header bar is shown (set by createNavigation).
    @Published var showHeader: Bool = false

    // Returns an error message on an invalid color (nil on success) so the caller
    // can surface it as a JS exception → error screen, mirroring Android's throw.
    func setBackgroundColor(_ raw: String) -> String? {
        guard let color = PropParser().parseColor(raw) else {
            return "invalid appBackgroundColor: \"\(raw)\" " +
                "(supported: #RGB | #RRGGBB | #RRGGBBAA | rgb() | rgba())"
        }
        backgroundColor = color
        return nil
    }

    func reset() {
        backgroundColor = AppConfig.defaultBackground
        showHeader = false
    }
}

File: packages/native-android/wrst-runtime/src/main/java/com/wrst/runtime/AppConfig.kt
package com.wrst.runtime

import androidx.compose.runtime.mutableStateOf
import androidx.compose.ui.graphics.Color
import com.wrst.runtime.parsers.PropParser

// App-wide config set from JS via createAppConfig(). Values live in Compose
// state so changes apply on hot reload (after the bundle re-evaluates).
object AppConfig {
    private val parser = object : PropParser() {}

    private val DEFAULT_BACKGROUND = Color.Black

    val backgroundColor = mutableStateOf(DEFAULT_BACKGROUND)

    fun setBackgroundColor(raw: String) {
        val color = parser.parseColor(raw)
        if (color == Color.Unspecified) {
            throw IllegalArgumentException(
                "invalid appBackgroundColor: \"$raw\" " +
                    "(supported: #RGB | #RRGGBB | #RRGGBBAA | rgb() | rgba())"
            )
        }
        backgroundColor.value = color
    }

    // Reset to defaults before each bundle load so a reload that omits
    // createAppConfig falls back instead of keeping the previous value.
    fun reset() {
        backgroundColor.value = DEFAULT_BACKGROUND
    }
}
